pkg/check/contentavailability/contentavailability.go

```diff
@@ -0,0 +1,178 @@
+	PostageAmount int64
+	PostageDepth  uint64
+	PostageLabel  string
+	Seed          int64
+}
+
+// NewDefaultOptions returns new default options.
+func NewDefaultOptions() Options {
+	return Options{
+		ContentSize:   1024 << 4,
+		GasPrice:      "",
+		PostageAmount: 1000,
+		PostageDepth:  16,
+		PostageLabel:  "test-label",
+		Seed:          0,
+	}
+}
+
+func (c *Check) Run(ctx context.Context, cluster *bee.Cluster, opts interface{}) (err error) {
+	o, ok := opts.(Options)
+	if !ok {
+		return fmt.Errorf("invalid options type")
+	}
+
+	rnd := random.PseudoGenerator(o.Seed)
+	idx := rnd.Intn(cluster.Size())
+	node := cluster.NodeNames()[idx]
+
+	content := make([]byte, o.ContentSize)
+	if _, err := rnd.Read(content); err != nil {
+		return fmt.Errorf("unable to create content: %w", err)
+	}
+
+	store := &putterMock{exists: make(map[string]swarm.Address)}
+	pipe := builder.NewPipelineBuilder(ctx, store, storage.ModePutUpload, false)
+	addr, err := builder.FeedPipeline(ctx, pipe, bytes.NewBuffer(content), int64(len(content)))
+	if err != nil {
+		return fmt.Errorf("unable to feed pipeline: %w", err)
+	}
+
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return err
+	}
+	client := clients[node]
+
+	overlays, err := cluster.FlattenOverlays(ctx)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("%s: %s\n", node, overlays[node])
+
+	batchID, err := client.GetOrCreateBatch(ctx, o.PostageAmount, o.PostageDepth, o.GasPrice, o.PostageLabel)
+	if err != nil {
+		return fmt.Errorf("node %s: unable to create batch id: %w", node, err)
+	}
+	fmt.Printf("node %s: batch id %s\n", node, batchID)
+
+	time.Sleep(5 * time.Second) // Wait for nodes to get notified of the batch..
+
+	contentAddr, err := client.UploadBytes(ctx, content, api.UploadOptions{BatchID: batchID})
+	if err != nil {
+		return fmt.Errorf("node %s: unable to upload content: %w", node, err)
+	}
+	fmt.Printf("node %s: content uploaded successfully: %s\n", node, addr)
+
+	time.Sleep(5 * time.Second) // Wait for nodes to sync.
+
+	isRetrievable, err := client.IsRetrievable(ctx, contentAddr)
+	if err != nil {
+		return fmt.Errorf("node %s: unable to check if content is retrievable: %w", node, err)
+	}
+	if !isRetrievable {
+		return fmt.Errorf("node %s: the uploaded content is not retrievable", node)
+	}
+	fmt.Printf("node %s: uploaded content is retrievable\n", node)
+
+	addresses := store.Addresses()
+	if len(addresses) == 0 {
+		return errors.New("empty list of addresses")
+	}
+
+	rmCount := 0
+	rmChAddr := addresses[len(addresses)-1]
+	for node, nClient := range clients {
+		if err := nClient.RemoveChunk(ctx, rmChAddr); err != nil {
+			return fmt.Errorf("node %s: unable to remove chunk %s: %w", node, rmChAddr, err)
+		}
+		fmt.Printf("node %s: chunk %s removed\n", node, rmChAddr)
+
+		if rmCount++; rmCount == len(clients) {
+			break
+		}
+
+		time.Sleep(5 * time.Second) // Wait for nodes to sync.
+
+		isRetrievable, err = client.IsRetrievable(ctx, contentAddr)
```

Comment:
i would execute this once after the for loop is over. this check inside the for loop could flake due to changes in connectivity patterns

Reply:
Removed.